src/payments/controllers/coupon.controller.ts

```
@@ -0,0 +1,150 @@
+import { CreateCouponDto, UpdateCouponDto } from '../dtos/create-coupon.dto';
+import { ValidateCouponDto, ValidateCouponResponseDto } from '../dtos/validate-coupon.dto';
+import { DiscountCoupon } from '../entities/discount-coupon.entity';
+import { PaymentContextType } from '../enums/payment.enums';
+
+@ApiTags('Coupons')
+@Controller('coupons')
+export class CouponController {
+  constructor(private readonly couponService: CouponService) {}
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new discount coupon' })
```

ESLint

[error] Replace `'Create·a·new·discount·coupon'` with `"Create·a·new·discount·coupon"`

(prettier/prettier)